internal/cmd/gocdk/internal/static/_assets/demo/demo_docstore.go

```diff
@@ -0,0 +1,257 @@
+	URL          string
+	Err          error
+	Create       bool
+	Key          string
+	Value        string
+	Revision     string
+	WriteSuccess bool
+}
+
+const (
+	docstoreTemplatePrefix = `
+<!DOCTYPE html>
+<html>
+<head>
+  <meta charset="UTF-8">
+  <title>gocloud.dev/docstore demo</title>
+</head>
+<body>
+  <p>
+    This page demonstrates the use of Go CDK's <a href="https://godoc.org/gocloud.dev/docstore">docstore</a> package.
+  </p>
+  <p>
+    It is currently using a docstore.Collection based on the URL "{{ .URL }}", which
+    can be configured via the environment variable "DOCSTORE_COLLECTION_URL".
+  </p>
+  <p>
+    See <a href="https://gocloud.dev/concepts/urls/">here</a> for more
+    information about URLs in Go CDK APIs.
+  </p>
+  <ul>
+    <li><a href="./list">List</a> the documents in the collection</li>
+    <li><a href="./edit?create=true">Create</a> a new document</li>
+</ul>
+{{if .Err}}
+  <p><strong>{{.Err}}</strong></p>
+{{end}}`
+
+	docstoreTemplateSuffix = `
+</body>
+</html>`
+
+	// Input: *docstoreBaseData.
+	docstoreBaseTemplate = docstoreTemplatePrefix + docstoreTemplateSuffix
+
+	// Input: *docstoreListData.
+	docstoreListTemplate = docstoreTemplatePrefix + `
+  {{range .Keys}}
+    <div>
+      <a href="./edit?key={{ . }}">{{ . }}</a>
+    </div>
+  {{end}}` + docstoreTemplateSuffix
+
+	// Input: *docstoreEditData.
+	docstoreEditTemplate = docstoreTemplatePrefix + `
+  {{if .WriteSuccess}}
+    Write succeeded!
+  {{else}}
+    <form method="POST">
+      <input type="hidden" name="create" value="{{ .Create }}">
+    {{if .Create}}
+      <p><label>
+        Enter a unique key:
+        <br/>
+        <input type="text" name="key">
+      </label></p>
+    {{else}}
+      <p>Update document with key <b>{{ .Key }}</b>:</p>
+      <input type="hidden" name="key" value="{{ .Key }}">
+    {{end}}
+    <p><label>
+      Value:
+      <br/>
+      <input type="text" name="value" value="{{ .Value }}">
+    </label></p>
+    <p>Revision: {{ .Revision }}</p>
+    <input type="submit" value="Write It!">
+    </form>
+  {{end}}` + docstoreTemplateSuffix
+)
+
+var (
+	docstoreBaseTmpl = template.Must(template.New("docstore base").Parse(docstoreBaseTemplate))
+	docstoreListTmpl = template.Must(template.New("docstore list").Parse(docstoreListTemplate))
+	docstoreEditTmpl = template.Must(template.New("docstore edit").Parse(docstoreEditTemplate))
+)
+
+func docstoreBaseHandler(w http.ResponseWriter, req *http.Request) {
+	input := &docstoreBaseData{URL: collectionURL}
+	if err := docstoreBaseTmpl.Execute(w, input); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+}
+
+// docstoreListHandler lists the items in a collection, possibly under a "prefix"
+// query parameter. Each listed directory is a link to list that directory,
+// and each non-directory is a link to view that file.
+func docstoreListHandler(w http.ResponseWriter, req *http.Request) {
+	input := &docstoreListData{URL: collectionURL}
```

Comment:
nit: it's odd that you call this `input` when it's the output.

Reply:
I guess it depends on how you look at it; it's the input to the template, but the output of this function. I'll rename in a separate PR.